## Overview of the sessions

You are now starting a series of eight computer sessions (or labs), each lasting four hours. The overarching goals of these labs are to familiarize yourselves with modern bioinformatic tools and best practices, and to illustrate-by-doing the most important topics discussed in class. By completing all sessions and the final project, you will get a good idea of the work of bioinformaticians and biologists today. To improve the flow and the logic of each step in the bioinformatic workflow, we have designed most labs to be connected to each other; however, should you have troubles to complete one lab, we will help you (for example by providing intermediate results) so you are not penalized for the following sessions.

The first two sessions are for you to learn the basic tools that you will need for the entire course: the UNIX system and command line, work on a cluster, bash scripting, Python etc. You will also start to manipulate some of the files and learn about file formats that you will use throughout the course (such as the Fasta format).

Sessions 3 to 5 focus on mitochondrial genomes. Mitochondrial genomes come in different flavours (circular, linear, large, small, etc), but for simplicity we selected short and circular genomes from a few model species of animals such as fruit fly, nematodes, mouse and great apes. You will learn how to circularize a mitochondrial genome, annotate it using different tools, and finally compare your genome to those of different species. These three sessions are based on a successful protocol used to teach genomics in US universities, see the full reference here: 'A CURE-based approach to teaching genomics using mitochondrial genomes, CourseSource 2019, Pogoda et al' and the accompanying document 'A guide to organellar genome assembly and annotation', Pogoda et al.

Sessions 6 to 8 form a project during which you will receive a main question to answer. To answer that question, you will have to select mitochondrial and autosomal sequences from relevant species and compare your sequences in appropriate ways, for example by building phylogenetic trees. Importantly, you will re-use data that you created during sessions 3 to 5.

You will have to work individually, as we know from experience that when two or more people work together on the same computer,only the active student is really engaged. However, we will encourage you to communicate with each other and help each other out, as we also know from experience that explaining something to someone else is a very efficient way of learning! To this end, we included several moments in the labs where you should discuss your results with colleagues.

We will use different ways of grading the labs - these will be described in the respective protocols. We encourage you to submit the reports as we progress through the sessions, so that you will get the most out of the labs and out of your teachers!

Finally, the labs you are working with this year are new or have been substantially modified compared to previous versions of this course. We did our best to make them informative and interesting, and for them to run smoothly. We welcome any comments and suggestions for improvement - do not hesitate to write them down as we go along as you might have forgotten a lot of comments by the time you fill the course evaluation! This will be very helpful for the teachers and future students. Thank you!

During the course, we will use resources from SNIC-UPPMAX in the form of a course project (storage and computing hours). We hereby thank SNIC-UPPMAX for these resources.


<!-- I think we should keep the same structure for all the labs, with for example: goal of the particular lab + how it fits with the other labs; background information if necessary; input (what they receive); outputs (what they create - which can also be input for other steps obviously); tools they will use; steps; format of report; something else? further reading? link to specific notions in the course? (both bioinformatics and genomics parts) In one of the later labs, we could even have them fill some of the steps e.g. output and tools as a task. -->

